add camera offset with mouse

KatsuroDev opened the pull request:
Camera now offsets with mouse position.

KatsuroDev committed:
Merge pull request #1 from Elatronion/develop

sync up fork with upstream

KatsuroDev committed:
add offset to camera with mouse

Camera now moves with the mouse offset.

Elatronion commented on godot-project/scenes/prefabs/Player/PlayerCamera.gd:
```diff
@@ -0,0 +1,7 @@
+
+
+
+func _process(delta):
+	offset_h = lerp(-0.5, 0.5, get_viewport().get_mouse_position().x / get_viewport_rect().size.x)
+	offset_v = lerp(-0.5, 0.5, get_viewport().get_mouse_position().y / get_viewport_rect().size.y)
```
It's a good idea that the PlayerCamera logic is separate from the Player, but this is also inconsistent and more importantly the camera is still movable during IMV cutscenes, which shouldn't be the case.
Either move the logic here to the player, or replicate and move the logic from player here.

Elatronion reviewed (changes requested):
I'd like to request that the movement effect is lessened.
Right now it's far too much, thanks! :)

Moreover, the camera logic is incomplete.

KatsuroDev committed:
fix camera offset when playing imv

Camera offset is reset to zero when playing imv, also not updated while doing so.
The camera offset is lessened as requested.

Elatronion commented:
Great job, I love the changes and the polish. Thanks! :)

Elatronion reviewed (approved):
The default value of the exported variable camrea_max_offset should probably be the one that's being used currently for the player, but it's fine I won't really touch it anyway.
